src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs

```diff
@@ -104,18 +106,18 @@ public static class LockFileUtils
                         else
                         {
                             AddAssets(aliases, library, package, targetGraph.Conventions, dependencyType, lockFileLib,
-                                framework, runtimeIdentifier, contentItems, nuspec, orderedCriteriaSets[i]);
+                                framework, runtimeIdentifier, contentItems, nuspec, orderedCriteriaSets[i].Item1);
```

Comment:
Now that we can name tuples, is it worth never introducing a new `Item1`, `Item2`, `Item3` etc?

Reply:
I don't really know how I can name it here given that we're not using `foreach` to get a name for the parameter. 

It's basically a value tuple inside a list and we're iterating over the list. 

We're explicitly not using foreach to save on allocations here.

Reply:
This syntax seems to work:
```c#
List<(List<SelectionCriteria> OrderedCriteria, bool FallbackUsed)> orderedCriteriaSets = cache.GetLabeledSelectionCriteria(targetGraph, framework);

...

fallbackUsed = orderedCriteriaSets[i].FallbackUsed;
```

Reply:
That worked. Thanks. I fixed it in the lockfilebuildercache file as well.